Frameworks/UIKit/UIImage.mm

```diff
@@ -215,348 +209,213 @@ + (id)imageWithCGImage:(CGImageRef)image scale:(float)scaleFactor orientation:(U
+        _scale = 1.0f;
+        _imageStretch.origin.x = 0.0f;
+        _imageStretch.origin.y = 0.0f;
+        _imageStretch.size.width = 1.0f;
+        _imageStretch.size.height = 1.0f;
     }
-
-    return ret;
+    return self;
 }
 
-static bool loadImageWithWICDecoder(UIImage* dest, REFGUID decoderCls, void* bytes, int length) {
-    IWICImagingFactory* pFactory = NULL;
-    IWICBitmapDecoder* pDecoder = NULL;
-    IStream* spStream = NULL;
-    IWICBitmapFrameDecode* pFrame = NULL;
-    HRESULT hr = S_OK;
-
-    MULTI_QI mq = { 0 };
-
-    mq.pIID = &IID_IWICImagingFactory;
-    hr = CoCreateInstanceFromApp(CLSID_WICImagingFactory, NULL, CLSCTX_INPROC_SERVER, NULL, 1, &mq);
-
-    if (SUCCEEDED(hr)) {
-        pFactory = (IWICImagingFactory*)mq.pItf;
-        hr = pFactory->CreateDecoder(decoderCls, NULL, &pDecoder);
+/**
+ @Status Interoperable
+*/
+- (instancetype)initWithCGImage:(CGImageRef)image {
+    if (!image) {
+        [self release];
+        return nil;
     }
 
-    if (SUCCEEDED(hr)) {
-        hr = ::CreateStreamOnHGlobal(NULL, FALSE, &spStream);
-        if (SUCCEEDED(hr)) {
-            ULONG written = 0;
-            hr = spStream->Write(bytes, length, &written);
-            if (SUCCEEDED(hr) && written == length) {
-                hr = pDecoder->Initialize(spStream, WICDecodeMetadataCacheOnLoad);
-
-                if (SUCCEEDED(hr)) {
-                    hr = pDecoder->GetFrame(0, &pFrame);
-                    if (SUCCEEDED(hr)) {
-                        if (!loadImageFromWICFrame(dest, pFactory, pFrame)) {
-                            TraceError(TAG, L"loadImageFromWICFrame failed");
-                            hr = E_FAIL;
-                        }
-                    } else {
-                        TraceError(TAG, L"IWICBitmapDecoder::GetFrame failed hr=%x", hr);
-                    }
-                } else {
-                    TraceError(TAG, L"IWICBitmapDecoder::Initialize failed hr=%x", hr);
-                }
-            } else {
-                TraceError(TAG, L"IStream::Write failed hr=%x len=%d written=%d", hr, length, written);
-            }
-        } else {
-            TraceError(TAG, L"CreateStreamOnHGlobal failed hr=%x", hr);
-        }
+    if (self = [self init]) {
+        m_pImage = image;
+        CGImageRetain(image);
     }
+    return self;
+}
 
-    if (pFactory) {
-        pFactory->Release();
+/**
+ @Status Interoperable
+*/
+- (instancetype)initWithCGImage:(CGImageRef)image scale:(float)scale orientation:(UIImageOrientation)orientation {
+    if (self = [self initWithCGImage:image]) {
+        _scale = scale;
+        _orientation = orientation;
     }
 
-    if (pDecoder) {
-        pDecoder->Release();
-    }
+    return self;
+}
 
-    if (spStream) {
-        spStream->Release();
+- (instancetype)_initWithCopyOfImage:(UIImage*)imageToCopy WithRenderingMode:(UIImageRenderingMode)renderingMode {
+    if(!imageToCopy) {
+        [self release];
+        return nil;
     }
```

Comment:
I'll leave it as is, given it's a check early.

---
In reply to: [84749628](https://github.com/Microsoft/WinObjC/pull/1185#discussion_r84749628) [](ancestors = 84749628)